### PR TITLE
Add workers pool to handle mongoose packets

### PR DESCRIPTION
**Pull request by aleklisi**
This PR introduces multiple workers for handling iqs in mod_pubsub.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[6132.1](https://travis-ci.org/esl/MongooseIM/jobs/485372199) / Erlang 19.3 / small_tests / 5473459dcf710c41b3982ab1dd1bbf0c34ba211c
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2202/6132/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2202/6132/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-01-28_12.35.31/lib.mongooseim.logs/run.2019-01-28_12.35.32/suite.log.html)

**Comment by mongoose-im** [comment]
[6134.1](https://travis-ci.org/esl/MongooseIM/jobs/485774428) / Erlang 19.3 / small_tests / 5dcdc9e8d4fe5432f27d49ee1dfcc63f2ca4ffe7
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2202/6134/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2202/6134/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-01-29_07.45.21/lib.mongooseim.logs/run.2019-01-29_07.45.22/suite.log.html)

File: src/pubsub/mod_pubsub.erl [hunk 1]
```diff
@@ -57,11 +57,8 @@
 -include("jlib.hrl").
 -include("pubsub.hrl").
 
--define(STDTREE, <<"tree">>).
--define(STDNODE, <<"flat">>).
--define(STDNODE_MODULE, node_flat).
--define(PEPNODE, <<"pep">>).
--define(PUSHNODE, <<"push">>).
+-define(HANDLERS_POOL_NAME, my_pool).
+-define(HANDLERS_POOL_SIZE, 5).
```
Comment (aleklisi) [inline src/pubsub/mod_pubsub.erl]: Extract it to config.

File: src/pubsub/mod_pubsub.erl [hunk 2]
```diff
@@ -283,8 +254,13 @@ init([ServerHost, Opts]) ->
             ok
     end,
 
-    ejabberd_router:register_route(Host, mongoose_packet_handler:new(?MODULE, self())),
     {_, State} = init_send_loop(ServerHost),
+
+    % TODO extract to supervision tree !!
+    wpool:start(),
+
+    {ok, Pid} = wpool:start_pool(?HANDLERS_POOL_NAME, [{workers, ?HANDLERS_POOL_SIZE}, {worker, {mod_pubsub_router, [State]}}]),
```
Comment (aleklisi) [inline src/pubsub/mod_pubsub.erl]: Start pool in a supervision tree.

**Comment by mongoose-im** [comment]
[6135.1](https://travis-ci.org/esl/MongooseIM/jobs/485800560) / Erlang 19.3 / small_tests / d27d2dd54ad3fba963d6d8f350f37485fefa565a
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2202/6135/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2202/6135/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-01-29_09.10.22/lib.mongooseim.logs/run.2019-01-29_09.10.23/suite.log.html)

**Comment by mongoose-im** [comment]
[6150.1](https://travis-ci.org/esl/MongooseIM/jobs/486837168) / Erlang 19.3 / small_tests / 577830c2cb33efd4e6665cc301ffef2e18d03f69
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2202/6150/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2202/6150/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-01-31_10.15.11/lib.mongooseim.logs/run.2019-01-31_10.15.12/suite.log.html)

**Comment by mongoose-im** [comment]
[6164.1](https://travis-ci.org/esl/MongooseIM/jobs/488433800) / Erlang 19.3 / small_tests / 578767e91589ac22151aa0b74e94ed2608dd759d
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2202/6164/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2202/6164/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-02-04_10.43.21/lib.mongooseim.logs/run.2019-02-04_10.43.22/suite.log.html)

**Comment by aleklisi** [comment]
This PR tried to address race conditions in mod pubsub mnesia but it didn't work.

**Pull request closed without merging**